### PR TITLE
Add search fields to gift card search

### PR DESCRIPTION
**Pull request by allisonlarson**
* Add recipient_email, order_number, and send_date search fields
* Perform search on index page
* Remove search results in the show page
* Remove show route
* Add purchased scope on gift cards
* Specs

### CODE REVIEW

File: app/controllers/spree/admin/gift_cards_controller.rb [hunk 1]
```diff
@@ -1,17 +1,19 @@
 class Spree::Admin::GiftCardsController < Spree::Admin::BaseController
-  before_filter :load_gift_card_history, only: [:show]
   before_filter :load_user, only: [:lookup, :redeem]
   before_filter :load_gift_card_for_redemption, only: [:redeem]
   before_filter :load_gift_card_by_id, only: [:edit, :update, :send_email]
   before_filter :load_order, only: [:edit, :update]
+  Spree::LineItem.whitelisted_ransackable_associations += %w[order]
```
Comment (allisonlarson) [inline app/controllers/spree/admin/gift_cards_controller.rb]: I'm not sure what the best way to do this actually is. I need the order association on the line item to be able to search by order number. This is not a good way to do it, so looking for a recommended better way.

File: app/views/spree/admin/gift_cards/index.html.erb [hunk 2]
```diff
@@ -1 +1,40 @@
...
+
+    <tbody>
+      <% @gift_cards.each do |gift_card| %>
+        <tr>
+          <td class="align-center"><%= gift_card.formatted_redemption_code %></td>
+          <td class="align-center"><%= number_to_currency(gift_card.amount) %></td>
+          <td class="align-center"><%= link_to "#{gift_card.purchaser.email}", edit_admin_user_path(gift_card.purchaser) %></td>
+          <td class="align-center"><%= gift_card.created_at.localtime.strftime("%F %I:%M%p") %></td>
+          <td class="align-center"><%= gift_card.recipient_email %></td>
+          <td class="align-center"><%= gift_card.formatted_send_email_at %></td>
```
Comment (philbirt) [inline app/views/spree/admin/gift_cards/index.html.erb]: Would this be `formatted_sent_at`? should we list both when we intend to send it and when it has been sent most recently?

File: app/views/spree/admin/gift_cards/index.html.erb [hunk 3]
```diff
@@ -1 +1,40 @@
...
+          <% if gift_card.redeemed? %>
+            <td class="align-center"><%= link_to "#{gift_card.redeemer.email}", edit_admin_user_path(gift_card.redeemer) %></td>
+            <td class="align-center"><%= gift_card.redeemed_at.localtime.strftime("%F %I:%M%p") %></td>
```
Comment (philbirt) [inline app/views/spree/admin/gift_cards/index.html.erb]: Can we push this into the model like all the other formatted_*_at methods? Should we override in spree-backend to make it EST?

File: app/views/spree/admin/gift_cards/index.html.erb (hunk 2 above)
Comment (allisonlarson) [inline app/views/spree/admin/gift_cards/index.html.erb]: Should there be two different fields, a search by the date it is supposed to be sent at and the date it was set?

File: app/controllers/spree/admin/gift_cards_controller.rb (hunk 1 above)
Comment (allisonlarson) [inline app/controllers/spree/admin/gift_cards_controller.rb]: @athal7 ^^

Comment (athal7) [inline app/controllers/spree/admin/gift_cards_controller.rb]: @allisonlarson usually we do those at the model level, since it is a global change

**Comment by philbirt** [comment]
Looks good to me :shipit:

**Pull request merged**